### ISSUE TITLE
GeolocateControl :disabled attribute not toggling when location services disabled

### ISSUE THREAD

**Issue by atomtay** [visgl/react-map-gl#1057]
Hello there! I'm not sure if I'm having a problem with the GeolocateControl React wrapper or more directly with Mapbox GL JS, so please let me know if I'm in the wrong place.

When I click my GeolocateControl button in Firefox 74 (or any browser where I've denied access to my location), the state of the button does not change. Upon examining the DOM tree, I found that the `disabled=""` attribute had not been added to the `.mapboxgl-ctrl-geolocate` button.

My implementation of Geolocate is as follows (full code for the component can be found [here](https://github.com/MAPC/trailmap3/blob/develop/app/javascript/pages/components/map.jsx), if helpful):
```
<ReactMapGL
    ref={this.mapRef}
    width="100vw"
    height="100vh"
    {...currentState.viewport}
    onViewportChange={(viewport) => {
        const { width, height, ...etc } = viewport;
        this.setState({ viewport: etc });
    }}
    mapboxApiAccessToken={process.env.MAPBOX_API_TOKEN}
    mapStyle={currentState.mapStyle}
>
    <GeolocateControl
         positionOptions={{ enableHighAccuracy: true }}
         trackUserLocation
         className="control-panel__geolocate" // this class just adds a custom icon and sets extra padding
      />
</ReactMapGL>
```

I've got React-Map-GL version 5.2.3 installed, and separately running the [Locate User](https://uber.github.io/react-map-gl/examples/locate-user) example locally on my computer works as expected. I'm using Firefox 74.0 on MacOS Catalina 10.15.3.

Happy to provide any additional information I may have missed. Thank you in advance!

**Comment by atomtay** [comment]
One small thing I've since discovered: I updated the Mapbox CSS stylesheet I was referencing (previously called 0.54.0, now calling 1.8.1), and when I manually add `disabled=""` to `.mapbox-ctrl-geolocate`, the appropriate styles appear. I also tried adding the geolocation control to a separate non-React Mapbox GL JS application and the `disabled=""` attribute came on as expected in Firefox.

**Comment by Pessimistress** [comment]
This will be addressed in v5.3/v6.1. Currently available in `6.1.0-beta.1`.

**Comment by yiziz** [comment]
i'm seeing this issue happen again
